clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -588,9 +581,9 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
     static constexpr uint64_t INSTRS_PER_US = 2000;
 
     // Called just once at initialization time to set the initial input-to-output
-    // mappings and state.
-    scheduler_status_t
-    set_initial_schedule(std::unordered_map<int, std::vector<int>> &workload2inputs);
+    // mappings and state for the particular mapping_t mode.
+    virtual scheduler_status_t
```

Comment:
Grouped all the virtual ones together.

Added:
    // Should call set_cur_input() for all outputs with initial inputs.

> Side question: how much effort do we want to put into making the interface between the driver scheduler and these derived schedulers clearer and cleaner?

Agreed that it can use more work.  Added to the issue:

The main goals of this initial round of refactoring are A) simplify the huge methods with if.else splits based on mode; B) split the code up better as scheduler.cpp was very very large.  The interfaces between the main class the subclasses could be more clearly separated and crystalized; that will likely be left for a future pass.

Reply:
Also added a comment in the new virtual method section of the header:

    /// XXX i#6831: These interfaces between the main class the subclasses could be
    /// more clearly separated and crystalized.